src/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs

```diff
@@ -12,40 +12,45 @@ namespace System.Net.Http
     public partial class HttpClientHandler : HttpMessageHandler
     {
         // This partial implementation contains members common to all HttpClientHandler implementations.
-        private const string ManagedHandlerSettingName = "COMPlus_UseManagedHttpClientHandler";
-        private const string AppCtxManagedHandlerSettingName = "System.Net.Http.UseManagedHttpClientHandler";
+        private const string ManagedHandlerEnvironmentVariableSettingName = "DOTNET_SYSTEM_NET_HTTP_USEMANAGEDHTTPCLIENTHANDLER";
+        private const string ManagedHandlerAppCtxSettingName = "System.Net.Http.UseManagedHttpClientHandler";
 
         private static LocalDataStoreSlot s_useManagedHandlerSlot;
 
         private static bool UseManagedHandler
         {
             get
             {
-                // Check the environment variable to see if it's been set to true.  If it has, use the managed handler.
-                if (Environment.GetEnvironmentVariable(ManagedHandlerSettingName) == "true")
+                // First check for the AppContext switch, giving it priority over over the environment variable.
+                if (AppContext.TryGetSwitch(ManagedHandlerAppCtxSettingName, out bool isManagedEnabled))
                 {
-                    return true;
+                    return isManagedEnabled;
                 }
 
-                if (AppContext.TryGetSwitch(AppCtxManagedHandlerSettingName, out bool isManagedEnabled) && isManagedEnabled)
+                // AppContext switch wasn't used. Check the environment variable to see if it's been set to true.
+                string envVar = Environment.GetEnvironmentVariable(ManagedHandlerEnvironmentVariableSettingName);
+                if (envVar != null && (envVar.Equals("true", StringComparison.OrdinalIgnoreCase) || envVar.Equals("1")))
                 {
                     return true;
                 }
 
+                // TODO #23166: Remove the following TLS check assuming the type is exposed publicly.  If it's not,
```

Comment:
Dumb question:  Why not just delete it now?  Is there a reason for keeping it?  Who is using it?

Reply:
We are.  We need a non-process-wide mechanism to enable the handler, e.g. we use it in ClientWebSocket, and when ClientWebSocket creates the HttpClientHandler it must use ManagedHandler.

Reply:
Don't those other places need to be updated then?

https://github.com/dotnet/corefx/search?utf8=%E2%9C%93&q=COMPlus_UseManagedHttpClientHandler&type=

Reply:
Yes, yes they do.